src/encoder/encodermp3.cpp

```diff
@@ -197,42 +223,56 @@ EncoderMp3::EncoderMp3(EncoderCallback* pCallback)
         return;
     }
     qDebug() << "Loaded libmp3lame version " << get_lame_version();
+    qDebug() << "lame_set_VBR_quality: " << QString((lame_set_VBR_quality == nullptr) ? "missing" : "present")
+        << " lame_encode_buffer_interleaved_ieee_float: " 
+        << QString((lame_encode_buffer_interleaved_ieee_float == nullptr) ? "missing" : "present")
+        << " id3tag_add_v2: " << QString((id3tag_add_v2 == nullptr) ? "missing" : "present");
 }
 
 // Destructor
 EncoderMp3::~EncoderMp3() {
-    if (m_library != NULL && m_library->isLoaded()) {
+    if (m_library != nullptr && m_library->isLoaded()) {
         flush();
         lame_close(m_lameFlags);
         m_library->unload(); //unload dll, so, ...
         qDebug() << "Unloaded libmp3lame ";
-        m_library = NULL;
+        m_library = nullptr;
     }
     //free requested buffers
-    if (m_bufferIn[0] != NULL)
+    if (m_bufferIn[0] != nullptr)
         delete m_bufferIn[0];
-    if (m_bufferIn[1] != NULL)
+    if (m_bufferIn[1] != nullptr)
         delete m_bufferIn[1];
-    if (m_bufferOut != NULL)
+    if (m_bufferOut != nullptr)
         delete m_bufferOut;
+}
 
-    lame_init =  0;
-    lame_set_num_channels = 0;
-    lame_set_in_samplerate =  0;
-    lame_set_out_samplerate = 0;
-    lame_close = 0;
-    lame_set_brate = 0;
-    lame_set_mode = 0;
-    lame_set_quality = 0;
-    lame_set_bWriteVbrTag = 0;
-    lame_encode_buffer_float = 0;
-    lame_init_params = 0;
-    lame_encode_flush = 0;
-
-    id3tag_init= 0;
-    id3tag_set_title = 0;
-    id3tag_set_artist = 0;
-    id3tag_set_album = 0;
+void EncoderMp3::setEncoderSettings(const EncoderSettings& settings)
+{
+    m_bitrate = settings.getQuality();
+    
+    int modeoption = settings.getSelectedOption(EncoderMp3Settings::ENCODING_MODE_GROUP);
+    m_encoding_mode = (modeoption==0) ? vbr_off : (modeoption==1) ? vbr_abr : vbr_default;
+
+    if (m_encoding_mode != vbr_off) {
+        if (lame_set_VBR_quality) {
+            // vbr range 0 to 9.999 (best to worst). slider range (0 to max) worse to best.
+            int val = settings.getQualityValues().size() - 1 - settings.getQualityIndex();
```

Comment:
Does this table match this: 
http://wiki.hydrogenaud.io/index.php?title=LAME#Recommended_settings_details